src/app/search.service.ts

```diff
@@ -108,7 +108,8 @@ export class SearchService {
   }
 
   hasTermFilterApplied(): boolean {
-    return (this.getQueryParams().term?.length > 0);
+    const term = this.getQueryParams().term || '';
+    return (term.length > 0);
```

Comment:
Not sure this adds much. The previous code was already type-safe.
```suggestion
    return (this.getQueryParams().term?.length > 0);
```

If possible can you please explain your reasoning for this change?

Reply:
The previous version produces an error with the new tsconfig. `length` could be undefined, so we can't use it as a number:

```

Error: src/app/search.service.ts:111:13 - error TS2532: Object is possibly 'undefined'.

111     return (this.getQueryParams().term?.length > 0);
                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```

Reply:
Ok. Thanks for explaining.